lib/house.rb

```diff
@@ -0,0 +1,74 @@
+require 'pry'
+require './lib/room'
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price.delete("$").to_i
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    rooms << room
+  end
+
+  def above_market_value?
+    if @price > 500000
+      true
+    else
+      false
+    end
+  end
+
+  def rooms_from_category(category)
+    rooms_from_cat = []
+    @rooms.each do |room|
+      if room.category == category
+        rooms_from_cat << room
+      end
+    end
+    return rooms_from_cat
+  end
+
+  def area
+    house_area = 0
+    @rooms.each do |room|
+      house_area += room.area
+    end
+    return house_area
+  end
+
+  def details
+    house_details = Hash.new
+    house_details["price"] = @price
+    house_details["address"] = @address
+    return house_details
+  end
+
+  def price_per_square_foot
+    (@price.to_f / area).round(2)
+  end
+
+  def rooms_sorted_by_area
+    sorted = []
+    sorted << rooms.sort_by do |room|
+      room.area
+    end
+    sorted.flatten!.reverse
+  end
```

Comment:
The reason you are having to use `flatten` is because you are kind of using both sort_by and the each pattern.  Take a look at this:

```ruby
  def rooms_sorted_by_area
    @rooms.sort_by do |room|
      room.area
    end.reverse
  end
```